src/machine/spi.go

```diff
@@ -0,0 +1,58 @@
+// +build nrf
+
+package machine
+
+import "errors"
+
+// Tx handles read/write operation for SPI interface. Since SPI is a syncronous write/read
+// interface, there must always be the same number of bytes written as bytes read.
+// The Tx method knows about this, and offers a few different ways of calling it.
+// This form sends the bytes in tx buffer, putting the resulting bytes read into the rx buffer.
+// Note that the tx and rx buffers must be the same size:
+// spi.Tx(tx, rx)
+// This form sends the tx buffer, ignoring the result. Useful for sending "commands" that return zeros
+// until all the bytes in the command packet have been received:
+// spi.Tx(tx, nil)
+// This form sends zeros, putting the result into the rx buffer. Good for reading a "result packet":
+// spi.Tx(nil, rx)
+func (spi SPI) Tx(w, r []byte) error {
+	if w == nil && r == nil {
+		return errors.New("SPI Tx requires a write or read slice, or both")
```

Comment:
Can you move this into a package global? At the moment the compiler is not smart enough to recognize this is a constant interface value, while it does recognize this when it is a global. This avoids a memory allocation when there is an error.